### DISCUSSION TITLE
Unable to identify "is" as a verb

### DISCUSSION THREAD

**Discussion by HSCLHD** [explosion/spaCy#8856] [Help: Coding & Implementations] [answered]
Hi there,
I just started with spaCy these days and was trying to extract the verb from a sentence. All worked fine but it failed to recognize "is" as a proper verb. I saw from the home page example that "was " is recognizable. The following is what i did and the code i used.
```
import spacy

nlp = spacy.load("en_core_web_sm")

text = (
        "Daniel is the star of the show."
)
doc = nlp(text)
print("Noun phrases:", [chunk.text for chunk in doc.noun_chunks])
print("Verbs:", [token.lemma_ for token in doc if token.pos_ == "VERB"])
```

Here is the log
```
Noun phrases: ['Daniel', 'the star', 'the show']
Verbs: []

```
returned an empty Verb list
I'm using 
macOS Big Sur 11.4 
spaCy version: not sure
Python version: Python 3.8.10 64-bit(base-conda)

**Comment by polm** [comment, marked as answer]
This is intended behavior. In Universal Dependencies, "to be" as used in this sentence is tagged as AUX in English. See [their docs](https://universaldependencies.org/u/pos/AUX_.html). (I do find it a bit surprising, but it is consistent with their examples.)

Couple of things:

- you can use `spacy info` at the command line to get your spaCy and model versions
- when you are confused by results, check the values of individual items (so in this case, see what the POS of "is" is)
- See this thread about model errors in general: #3052

**Reply by HSCLHD** [reply]
Thanks polm for your reply. It makes sense now. 
I used the tag_ and pos_ as suggested, returns AUX VBD. Guess have to include aux when defining the return.
also there seems to be an error on the output of the homepage. The attatched is the screenshot. 'was' is identified as a verb in the example.
<img width="696" alt="截屏2021-08-07 11 37 36" src="https://user-images.githubusercontent.com/77868814/128586726-1657348c-5ba4-4f28-b09e-c63bcd575001.png">

**Reply by polm** [reply]
Please don't post screenshots of code, just copy/paste the text so we can run it ourselves. 

I don't think that's an error either, "to be" can be a VERB or AUX depending on the situation, please see the Universal Dependencies docs. 

Also please note as described in the linked thread that the models are just wrong sometimes, so you have to be prepared to deal with that, and we can't apply fixes to isolated errors in specific sentences, the technology doesn't work that way.

**Reply by HSCLHD** [reply]
Hi polm. Thanks for the reply.

I understand there is no such thing as silver bullet and all tools and codes need to be adjusted and refined before final implementation.

I copied the code and ran it locally but it didn't return the same result as demonstrated on the homepage. The reason for the screenshot is to send the feedback that the demonstration code didn't perform as expected. I appreciate the hard work of the team and the strong functionality spaCy is able to provide. Only hope that the front page example can change into another one so that there would be less ambiguity.

**Reply by adrianeboyd** [reply]
Actually, this may point to an error in the rule-based tag->pos conversion in the `en` pipelines. In spacy v2, all forms of "to be" were converted to `AUX`, but in v3 if the dependency parse is available, it tries to distinguish `AUX` and `VERB` uses. The rule currently only considers `ROOT` and `ccomp` and it probably should also consider "be" when it's the root of an `advcl` clause. We will take a look!

**Reply by adrianeboyd** [reply]
I looked into the details and the answer is pretty much: yikes, we're probably not going to be able to do a very good good job of conforming to the UD guidelines with our rule-based conversion from tag->pos in the pretrained pipelines with the ClearNLP OntoNotes dependency trees rather than UD dependencies. And there is not even consensus on all the details between the UD English corpora for certain cases.

The underlying problem in our pipelines is that there's no AUX/VERB distinction in the underlying PTB tags, so you have the information that the word/tag `is/VBZ` and this may be `is/AUX` or `is/VERB` with UD POS.

Clear cases are if it's `aux` or `auxpass` in the parse. Cases where it's the root of a clause are harder because existential "be" should be `VERB` (and would be the root in UD) and copula "be" is `AUX` and `cop` in most other cases (so not the root in UD). In the ClearNLP parses, "be" is the root in both of these cases and attribute ruler does not have enough information to distinguish these. You could probably do fairly well by looking for an immediately preceding `there` to catch existential cases, although the dependency matcher would do a better job.

Some UD details: https://universaldependencies.org/u/overview/simple-syntax.html#english. According the UD standards, 1b and 6 are `VERB` while the others are `AUX`. There is disagreement about the analysis in 1b.

Right now we only convert `VERB` -> `AUX` if its `ROOT` or `ccomp` and we should probably add `advcl`, `relcl`, and `xcomp`, and then add exceptions for the straightforward cases with existentials. I will look into it for the next English model releases.

Overall, given the constraints of the original treebank and training data, the `AUX`/`VERB` distinction is probably never going to be that good in the output of the current English pipelines, so you may be better off relying on the PTB tags (`token.tag`) or treating `AUX`/`VERB` as equivalent for most use cases.
